File: phoenix/keycloakuser/src/main/java/com/phoenix/dtostock/SoldProductDto.java
package com.phoenix.dtostock;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class SoldProductDto {
    private String serialNumber;
    private String simNumber;
    private LocalDate soldDate;
    private boolean checkedSell;
    private String productType;
    private String brand;
    private String prodName;
    private String comments;
    private BigDecimal price;
    private StockDto stock;
    private AgentProdDto managerSoldProd;
    private AgentProdDto agentWhoSold;
    private AgentProdDto agentAssociatedProd;
}

File: phoenix/phoenixfront/src/app/models/inventory/SoldProductDto.ts
import { AgentProdDto } from "./AgentProdDto";
import { Stockdto } from "./Stock";

export interface SoldProductDto {
    serialNumber?: string;
    simNumber?: string;
    checkout?: Date;
    checkedSell?: boolean;
    brand?: string;
    productType?: string;
    prodName?: string;
    comments?: string;
    price?: number;
    soldDate?: Date;
    stock?: Stockdto;
    managerSoldProd?: AgentProdDto;
    agentAssociatedProd?: AgentProdDto;
    agentWhoSold?: AgentProdDto;

}